lab_5_scrapper/scrapper.py

```diff
@@ -90,75 +214,151 @@ def __init__(self, config: Config) -> None:
         """
         Returns seed_urls param
         """
-        pass
+        return self.config.get_seed_urls()
 
 
 class HTMLParser:
     """
     ArticleParser implementation
     """
 
-    def __init__(self, full_url: str, article_id: int, config: Config) -> None:
+    def __init__(self, article_url: str, article_id: int, config: Config) -> None:
         """
         Initializes an instance of the HTMLParser class
         """
-        pass
+        self.article_url = article_url
+        self.article_id = article_id
+        self.config = config
+        self.article = Article(url=self.article_url, article_id=self.article_id)
 
     def _fill_article_with_text(self, article_soup: BeautifulSoup) -> None:
         """
         Finds text of article
         """
-        pass
+        # get the title
+        title = article_soup.find_all('h1', {'class': 'new-title'})[0]
+        title_text = title.text
+
+        # get the preview
+        preview = article_soup.find_all('div', {'class': 'preview-text'})[0]
+        preview_text = preview.text.strip()
+        if preview_text[-1] not in '.?!':
+            preview_text += '.'
+
+        # get the article body
+        body_bs = article_soup.find_all('div', {'class': 'detail'})[0]
+        paragraph_texts = [par.text.strip() for par in body_bs.find_all('p')]
+        article_text = ' '.join(paragraph_texts)
+
+        self.article.title = title_text
+        self.article.text = preview_text + ' ' + article_text
 
     def _fill_article_with_meta_information(self, article_soup: BeautifulSoup) -> None:
         """
         Finds meta information of article
         """
-        pass
+        try:
+            author = article_soup.find_all('span', {'class': 'author'})
+            self.article.author.extend(author[0].text[7:].split(', '))
+        except IndexError:
+            self.article.author.append('NOT FOUND')
+
+        article_date = article_soup.find(itemprop='datePublished').get('content')
+        article_time = article_soup.find_all('span', {'class': 'date'})[0].text[-6:]
+        date_n_time = self.unify_date_format(article_date + article_time)
+        self.article.date = date_n_time
 
     def unify_date_format(self, date_str: str) -> datetime.datetime:
         """
         Unifies date format
         """
-        pass
+        return datetime.datetime.strptime(date_str, '%Y-%m-%d %H:%M')
 
     def parse(self) -> Union[Article, bool, list]:
         """
         Parses each article
         """
-        pass
+        response = make_request(self.article_url, self.config)
+        article_bs = BeautifulSoup(response.text, 'lxml')
+        self._fill_article_with_text(article_bs)
+        self._fill_article_with_meta_information(article_bs)
+        return self.article
 
 
 def prepare_environment(base_path: Union[Path, str]) -> None:
     """
     Creates ASSETS_PATH folder if no created and removes existing folder
     """
-    pass
+    try:
```

Comment:
instead of exception, check for existence using standard API

Reply:
done